### PR TITLE
tests: Move AWS VPC tests to RSpec

### PR DESCRIPTION
**Pull request by mxinden**
Previously we were bringing up a VPC via bash scripts. This commit
replaces the existing bash scripts with Ruby code.

### CODE REVIEW

**Comment by mxinden** [comment]
ok to test

File: Jenkinsfile [hunk 1]
```diff
@@ -134,12 +134,28 @@ pipeline {
...
+            node('worker && ec2') {
+              withCredentials(creds) {
+                checkout scm
+                unstash 'installer'
+                unstash 'smoke'
+                withDockerContainer(
+                    image: tectonic_smoke_test_env_image,
+                    args: '--device=/dev/net/tun --cap-add=NET_ADMIN -u root'
+                ) {
+                  sh """#!/bin/bash -ex
```
Comment (cpanato) [inline Jenkinsfile]: @mxinden can you please add the `ansiColor('xterm')`?

**Comment by mxinden** [comment]
@cpanato Thanks for the hint. Any other suggestions?

**Comment by mxinden** [comment]
ok to test

**Comment by mxinden** [comment]
@cpanato great catch! Adjusted accordingly.

**Comment by mxinden** [comment]
I will merge here. The previous build (# 3) succeeded entirely. # 4 failed in unrelated stages (bash aws and azure).

**Comment by mxinden** [comment]
@cpanato Thanks for the review.

**Pull request merged**